### ISSUE TITLE
Unclear Screen Reader Focus When Cancelling Deletion in Danger Zone

### ISSUE THREAD

**Issue by kevinandrews1** [bbertucc/equalify#354]
When closing or cancelling a deletion action, the screen reader focus is unclear. Focus should return to the button that initiated the dialog, i.e., the delete button. After cancelling the deletion action, the screen reader does not clearly indicate where the focus has shifted, causing confusion for users. This issue occurs in multiple areas, not just in the Reports section. To improve accessibility, the focus should be explicitly set back to the trigger button (the delete button) that initiated the deletion dialog. This ensures a logical and intuitive navigation flow for users relying on screen readers throughout the application. closing and canceling do the same thing so there should be just one or the other.

**Comment by wilsuriel03** [comment]
Complete ✅

**Comment by kevinandrews1** [comment]
@wilsuriel03 I'm still running into this. When you cancel deletion and then start to navigate, i.e. with the tab key, the first item you land on is the go to homepage link in the banner nav area and basically have to traverse the entire page and try to recover your place.

**Comment by wilsuriel03** [comment]
we may need to visit this on monday , and would be great if you can demo the issue. It seems to be working when I use tab.
@kevinandrews1

**Comment by kevinandrews1** [comment]
@wilsuriel03 here's the [issue](https://drive.google.com/file/d/1EZ5CB98GMPKErw9Y-JiswsAt6Z9G0A_D/view?usp=drive_link) as I'm currently seeing it.

**Comment by wilsuriel03** [comment]
@bbertucc  @heythisischris  what do you guys suggest to fix this issue ? its only not working for one screen reader if you pay attention to the demo that Kevin posted.

**Comment by bbertucc** [comment]
@wilsuriel03 - you should add some JS that manages the focus area when "Close" or cancel is clicked. Basically, you need to make it so closing the modal or clicking cancel leads back to the button that launched the modal. Here is a dumb example I created with ChatGPT: https://chatgpt.com/share/901006c4-9ed1-4b61-87b7-30392ae61490


Also note, you probably should get rid of that close "x" because it's redundant for screen readers. Users should only be able to close by clicking "Cancel."

**Comment by bbertucc** [comment]
O @wilsuriel03 - you can also use the native `<dialog>` to achieve this. Check out this post: https://www.matuzo.at/blog/2023/focus-dialog/

**Comment by bbertucc** [comment]
Looks like the native radix dialogue works as expected. If you click "Edit Profile" with your keyboard navigation, then close the dialogue, it goes back to "Edit Profile" here: https://www.radix-ui.com/primitives/docs/components/dialog

**Comment by kevinandrews1** [comment]
@wilsuriel03 and I met on this one this afternoon and we have a good path forward. Also mentioned removing the duplciate cancel/close.

**Comment by wilsuriel03** [comment]
@bbertucc this is fixed and ready to retest @kevinandrews1

**Comment by bbertucc** [comment]
Sweet! I switched the asignee to @kevinandrews1 for testing
